autobahn/autobahn/wamp/protocol.py

```diff
@@ -432,41 +445,25 @@ def error(err):
             elif isinstance(msg, message.Event):
 
                 if msg.subscription in self._subscriptions:
-
                     handler = self._subscriptions[msg.subscription]
-
                     if handler.details_arg:
                         if not msg.kwargs:
                             msg.kwargs = {}
                         msg.kwargs[handler.details_arg] = types.EventDetails(publication=msg.publication, publisher=msg.publisher, topic=msg.topic)
 
+                    invoke_args = (handler.obj,) if handler.obj else tuple()
+                    if msg.args:
+                        invoke_args = invoke_args + tuple(msg.args)
+                    invoke_kwargs = msg.kwargs if msg.kwargs else dict()
                     try:
-                        if handler.obj:
-                            if msg.kwargs:
-                                if msg.args:
-                                    handler.fn(handler.obj, *msg.args, **msg.kwargs)
-                                else:
-                                    handler.fn(handler.obj, **msg.kwargs)
-                            else:
-                                if msg.args:
-                                    handler.fn(handler.obj, *msg.args)
-                                else:
-                                    handler.fn(handler.obj)
-                        else:
-                            if msg.kwargs:
-                                if msg.args:
-                                    handler.fn(*msg.args, **msg.kwargs)
-                                else:
-                                    handler.fn(**msg.kwargs)
-                            else:
-                                if msg.args:
-                                    handler.fn(*msg.args)
-                                else:
-                                    handler.fn()
+                        handler.fn(*invoke_args, **invoke_kwargs)
 
                     except Exception as e:
+                        self.onUserError(e)
```

Comment:
What if `onUserError` raises? Probably just eat any exception then without any logging or whatever would be fine ..